### ISSUE TITLE
unicode characters in non-existent command names result in mutated output

### ISSUE THREAD

**Issue by thecodejunkie** [dotnet/cli#2621]
## Steps to reproduce

Type `dotnet <invalid command>` in the CLI
## Expected  behavior

An error output without scrambled text
## Actual behavior

```
dotnet buiöd
←[1m←[31mNo executable found matching command "dotnet-bui├â┬Âd"←[39m←[22m
```

```
λ dotnet bud
←[1m←[31mNo executable found matching command "dotnet-bud"←[39m←[22m
```
## Environment data

`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-rc2-002457)

Product Information:
 Version:     1.0.0-rc2-002457
 Commit Sha:  d06418509d

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.3.9600
 OS Platform: Windows
 RID:         win81-x64
```

**Comment by khellang** [comment]
I could've sworn this was fixed with #1212...

**Comment by Bartmax** [comment]
is weird:)

powershell first try, error described above.
run cmd, no scramble
exit to ps and now no scramble :)

![image](https://cloud.githubusercontent.com/assets/3386797/14702675/12f1346a-0782-11e6-80e6-16eb446129ee.png)

**Comment by thecodejunkie** [comment]
@khellang Regression, yo? :D

**Comment by piotrpMSFT** [comment]
There are two issues here. One is that unicode characters are being scrambled when written back out from CLI:
````
dotnet buiöd --> dotnet-bui├â┬Âd
````

The other issue is that folks' command lines are somehow getting into a state where they are not respecting ANSI Color codes. We're tracking this one here: https://github.com/dotnet/cli/issues/2590

Can folks comment on that issue to help us get to a repro? We're trying exactly the same recipe on Win10, running `dotnet foo`, both from cmd and ps, and we're only seeing colored output and not control characters.

I'm repurposing this bug to track the abovementioned Unicode issue.

**Comment by livarcocc** [comment]
The unicode character issue seems to be fixed with the latest 2.0 CLI.
